### ISSUE TITLE
Valid CER going back to 1.0 & all samples in this mini-batch got nan grad.

### ISSUE THREAD

**Issue by kbramhendra** [espnet/espnet#4267]
Hi,
    I am training conformer AED model with phonemes, The model trained till 8 epochs after that the model valid cer is going to 1.0. and in the training log showing "All samples in this mini-batch got nan grad. Returning nan value instead of CTC loss, can you please help
!
![cer](https://user-images.githubusercontent.com/103481448/162886316-1f6bc9c5-52e0-4b0b-9975-f873da6e1ec7.png)

"

**Comment by sw005320** [comment]
Which recipe and model are you using?

**Comment by kbramhendra** [comment]
This is on custom data, the same data is working on bpe5000 but showing like this for phonemes

**Comment by sw005320** [comment]
which model are you using?
can you paste the config file?

**Comment by kbramhendra** [comment]
encoder: conformer
encoder_conf:
    output_size: 512
    attention_heads: 4
    linear_units: 1024
    num_blocks: 12
    dropout_rate: 0.1
    positional_dropout_rate: 0.1
    attention_dropout_rate: 0.1
    input_layer: conv2d
    normalize_before: true
    macaron_style: true
    rel_pos_type: latest
    pos_enc_layer_type: rel_pos
    selfattention_layer_type: rel_selfattn
    activation_type: swish
    use_cnn_module: true
    cnn_module_kernel: 31

decoder: transformer
decoder_conf:
    attention_heads: 4
    linear_units: 1024
    num_blocks: 6
    dropout_rate: 0.1
    positional_dropout_rate: 0.1
    self_attention_dropout_rate: 0.1
    src_attention_dropout_rate: 0.1

model_conf:
    ctc_weight: 0.3
    lsm_weight: 0.1
    length_normalized_loss: false

frontend_conf:
    n_fft: 512
    win_length: 400
    hop_length: 160

seed: 2022
log_interval: 400   
num_att_plot: 0     
num_workers: 4      
sort_in_batch: descending       # how to sort data in making batch
sort_batch: descending          # how to sort created batches
batch_type: numel
batch_bins: 2000000
accum_grad: 30
max_epoch: 30
patience: none
init: none
best_model_criterion:
-   - valid
    - acc
    - max
keep_nbest_models: 10

use_amp: true      
cudnn_deterministic: false  
cudnn_benchmark: false      

optim: adam
optim_conf:
    lr: 0.0002
    weight_decay: 0.000001
scheduler: warmuplr
scheduler_conf:
    warmup_steps: 15000

specaug: specaug
specaug_conf:
    apply_time_warp: true
    time_warp_window: 5
    time_warp_mode: bicubic
    apply_freq_mask: true
    freq_mask_width_range:
    - 0
    - 27
    num_freq_mask: 2
    apply_time_mask: true
    time_mask_width_ratio_range:
    - 0.
    - 0.05
    num_time_mask: 5
ctc_conf:
    ignore_nan_grad: true

**Comment by sw005320** [comment]
How many hours for the training data?
Regardless of the amount of the training data, I recommend you tune `lr: 0.0002`

**Comment by kbramhendra** [comment]
Sure Thank you for the swift reply. The training data is about 2k hours. Is there any significance of accum_grad parameter nan grad outcome.

**Comment by sw005320** [comment]
Oh, I see. 
How many distinct phonemes do your corpus have?

**Comment by kbramhendra** [comment]
There are 73 phones, here is the list for your reference 


[tokens.txt](https://github.com/espnet/espnet/files/8472880/tokens.txt)

**Comment by kbramhendra** [comment]
Hi,
    I have experimented with different learning rates lr, the model is training for some epochs and throwing this warning  "The grad norm is nan. Skipping updating the model." causing to fail the model completely.
